pkgs/development/python-modules/openai/default.nix

```diff
@@ -0,0 +1,41 @@
+{ lib
+, buildPythonPackage
+, fetchPypi
+, pythonOlder
+
+# Python dependencies
+, openpyxl
+, pandas
+, pandas-stubs
+, requests
+, tqdm
+}:
+
+buildPythonPackage rec {
+  pname = "openai";
+  version = "0.11.0";
+
+  disabled = pythonOlder "3.7.1";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "d3e7fecd728e70fa9ca3484a00ea3f7b928b172207c6017816419d57f2b3b596";
+  };
+
+  propagatedBuildInputs = [
+    openpyxl
+    pandas
+    pandas-stubs
+    requests
+    tqdm
+  ];
+
+  # Tests aren't included in package
+  doCheck = false;
```

Comment:
There are tests in the upstream repo. Using the source GitHub could allow to run the tests.

Reply:
My main concern with using the GitHub source would be that the latest release version there (0.10.5) is older than the version on PyPi (0.11.0).

Reply:
we can fetch https://github.com/openai/openai-python/commit/62f8d40ffc6d3a723f5f1d99fee7febfd47026b5

Reply:
Done.